### ISSUE TITLE
RocksJava builds are broken due to bzip.org domain ownership issues

### ISSUE THREAD

**Issue by sagar0** [facebook/rocksdb#4305]
RocksJava static builds from source (using `make rocksdbjavastatic`) are currently failing with the below error, on all branches which depend on downloaded bzip sources.
```
~/rocksdb  $ make rocksdbjavastatic -j100
....
...
rm -rf bzip2-1.0.6
curl -O -L http://www.bzip.org/1.0.6/bzip2-1.0.6.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   271  100   271    0     0   1062      0 --:--:-- --:--:-- --:--:--  1062
BZIP2_SHA256_ACTUAL=`sha256sum bzip2-1.0.6.tar.gz | cut -d ' ' -f 1`; \
if [ "a2848f34fcd5d6cf47def00461fcb528a0484d8edef8208d6d2e2909dc61d9cd" != "$BZIP2_SHA256_ACTUAL" ]; then \
        echo bzip2-1.0.6.tar.gz checksum mismatch, expected=\"a2848f34fcd5d6cf47def00461fcb528a0484d8edef8208d6d2e2909dc61d9cd\" actual=\"$BZIP2_SHA256_ACTUAL\"; \
        exit 1; \
fi
bzip2-1.0.6.tar.gz checksum mismatch, expected="a2848f34fcd5d6cf47def00461fcb528a0484d8edef8208d6d2e2909dc61d9cd" actual="e75fb19a6e167b6bf5b614e9ca4df333f9b3bd06edb5ea0a7cb60d19cb789065"
make: *** [libbz2.a] Error 1
```
I found that bzip.org is no longer being maintained, and RocksJava depends on bzip2 packages downloaded from bzip.org. 

Our Makefile showing the relevant code:
https://github.com/facebook/rocksdb/blob/f1f5ba085fa28e3dbcd5202d6451d2b98a1492a9/Makefile#L1622-L1624
https://github.com/facebook/rocksdb/blob/f1f5ba085fa28e3dbcd5202d6451d2b98a1492a9/Makefile#L1681-L1688

I couldn't find any other credible source to get the bzip packages from ... yet. 
This seems to be actively discussed on lwn as well, since Aug 9th: https://lwn.net/Articles/762264/

**Comment by sagar0** [comment]
I suggest to temporarily get the packages using wayback machine, as someone suggested in that lwn article link above. 

A link to most recent snapshot of bzip 1.0.6 tar.gz taken on June 6th 2018: https://web.archive.org/web/20180624184835/http://www.bzip.org/1.0.6/bzip2-1.0.6.tar.gz
(got from the archives as shown here: https://web.archive.org/web/*/http://www.bzip.org/1.0.6/bzip2-1.0.6.tar.gz) .

**Comment by sagar0** [comment]
This is temporarily fixed in #4306 by switching the download from web.archive.org, but we should find a long term solution.
